web/src/beta/features/Editor/Tabs/Storytelling/StoryPageIndicator/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { FC } from "react";
+
+import { styled } from "@reearth/services/theme";
+
+import StoryPageIndicatorItem from "./StoryPageIndicatorItem";
+
+type Props = {
+  currentPage: number;
+  currentPageProgress: number;
+  maxPage: number;
+  onChangePage: (page: number) => void;
+};
+
+const StoryPageIndicator: FC<Props> = ({
+  currentPage,
+  currentPageProgress,
+  maxPage,
+  onChangePage,
+}) => {
+  return (
+    <Wrapper>
+      {Array.from({ length: maxPage }).map((_, i) => {
+        const page = i + 1;
+        const isActive = currentPage >= page;
+        const isCurrentPage = page === currentPage;
+
+        return (
+          <StoryPageIndicatorItem
+            key={i}
+            page={page}
+            progress={isCurrentPage ? currentPageProgress : isActive ? 100 : 0}
+            onChangePage={onChangePage}
+          />
+        );
+      })}
+    </Wrapper>
+  );
```

Comment:
```suggestion
return (
    <Wrapper>
      {[...Array(maxPage)].map((_, i) => {
        const page = i + 1;
        const isActive = currentPage >= page;
        const progress = isActive ? 100 : 0;
        const isCurrentPage = page === currentPage;
        const indicatorProps = {
          key: i,
          page,
          progress: isCurrentPage ? currentPageProgress : progress,
          onChangePage,
        };

        return <StoryPageIndicatorItem {...indicatorProps} />;
      })}
    </Wrapper>
  );
```

Using [...Array(maxPage)] creates an array of length maxPage filled with undefined elements, which is then passed to map for iteration.

Used spread syntax (...) to pass the indicatorProps object as props to the StoryPageIndicatorItem component. This makes the code cleaner and eliminates repetitive prop assignments.

Reply:
Thank you, I applied `[...Array(maxPage)]`.

Sorry I couldn't get meaning of `eliminates repetitive prop assignments` ... :pray:

If you define the prop object first like your suggestion
- We lose type completion ( of course we can make it work by adding type )
- If return value will be bigger like `<A><B><C><D></A></B></C></D>`, we might not keep this strategy. So I feel weird.

Reply:
By using the spread syntax ({...}), we can pass all the properties of indicatorProps as separate props to the StoryPageIndicatorItem component. This eliminates the need to assign each prop individually.


In that case, you can explicitly specify the type of the indicatorProps object to maintain type checking and completion. For example:

```
const indicatorProps: StoryPageIndicatorItemProps = {
  key: i,
  page,
  progress: isCurrentPage ? currentPageProgress : progress,
  onChangePage,
};
```
By defining StoryPageIndicatorItemProps as the type for indicatorProps, you can ensure type safety and maintain proper type completion.


If the return value becomes more complex with nested components like <A><B><C><D></A></B></C></D>, it might not be suitable to use this strategy. In such cases, you can choose to keep the individual prop assignments to ensure clarity and maintainability of the code structure.


So I guess, Its upto you. Its not necessary to use them as `objectProps`..